### PR TITLE
Add docs for test parameter need to be named `t`

### LINKED ISSUE #1031: Docs: Really, use `t` for assertions
I just tried to covert a project that's tested with [tape](https://github.com/substack/tape) over to using AVA instead. The tests all follow this kind of format:

``` js
test('function-a', (assert) => {
  assert.plan(1);

  assert.is(
    functionA(),
    resultB,
    'function a should return result b'
  );
});
```

They seem to run just fine under AVA, but the assertion messages don't show up when I create a failing test case, just the test messages.

Is there a way to report on the assertion messages? Or is the best practice just to wrap every one of these assertions in its own `test` block? (In which case, why do the assertions accept optional message parameters?

Am I missing something stupid?

### PR DESCRIPTION
**Pull request by gurpreetatwal**
- Reworded the statement in the readme regarding the first parameter needing to be `t` to make it stronger and explain why its required.
- Add a note in common-pitfalls regarding the same issue
- Make the indentation style consistent in the code samples 

Closes #1031

### CODE REVIEW

File: docs/common-pitfalls.md [hunk 1]
```diff
@@ -63,6 +63,16 @@ test(async t => {
 
 AVA [can't trace uncaught exceptions](https://github.com/avajs/ava/issues/214) back to the test that triggered them. Callback-taking functions may lead to uncaught exceptions that can then be hard to debug. Consider promisifying and using `async`/`await`, as in the above example. This should allow AVA to catch the exception and attribute it to the correct test.
 
+### Why is the power-assert information not shown?
+
+Ensure that the first parameter passed into your test is named `t`. This is a byproduct of the way power-assert works. It uses a pattern matching scheme that makes it easier for implementors to wrap any assertion library with power-assert goodness without having to understand the ES AST at all. See [#1031](https://github.com/avajs/ava/issues/1031) for more details.
```
Comment (vdemedes) [inline docs/common-pitfalls.md]: > This is a byproduct of the way power-assert works.

I'd rather say that "This is a requirement by power-assert.".

**Comment by vdemedes** [comment]
Good job explaining the what & why!

File: docs/common-pitfalls.md (hunk 1 above)
Comment (novemberborn) [inline docs/common-pitfalls.md]: Could you change this to:

> ### Why are the enhanced assertion messages not shown?
>
> Ensure that the first parameter passed into your test is named `t`. This is a requirement of how AVA uses [`power-assert`](https://github.com/power-assert-js/power-assert), the library that provides the enhanced messages. works.

I see that this paragraph originally comes from @jamestalmage's comment in https://github.com/avajs/ava/issues/1031#issuecomment-244474530. It's quite technical though. I don't think we need that kind of detail here.

File: docs/common-pitfalls.md [hunk 2]
```diff
@@ -63,6 +63,16 @@ test(async t => {
...
+
+```js
+test(t => {
+	t.is(1, 1);
+}
```
Comment (novemberborn) [inline docs/common-pitfalls.md]: Missing `);` at the end here.

File: readme.md [hunk 3]
```diff
@@ -74,13 +74,13 @@ Your `package.json` will then look like this:
...
-  "devDependencies": {
-    "ava": "^0.15.0"
-  }
+	"name": "awesome-package",
+	"scripts": {
+		"test": "ava"
+	},
+	"devDependencies": {
+		"ava": "^0.15.0"
+	}
```
Comment (novemberborn) [inline readme.md]: Could you revert these to their original two space indentation? It's how `package.json` files are formatted.

File: docs/common-pitfalls.md (hunk 1 above)
Comment (gurpreetatwal) [inline docs/common-pitfalls.md]: I fully agree, that was just some blind copy and pasting from the comment on my part. Should I still keep the link to the issue for anyone who wants to know more?

Comment (novemberborn) [inline docs/common-pitfalls.md]: Nah, the issue still requires a lot of careful reading and background to fully understand why it is this way.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**